src/delagent/agent/delagent.c

```diff
@@ -157,7 +157,7 @@ int main (int argc, char *argv[])
     DeleteUpload(DelUpload); 
     fprintf(stdout, "The upload '%ld' is deleted by the user '%s'.\n", DelUpload, user_name);
   }
-  if (DelFolder) { DeleteFolder(DelFolder); }
+  if (DelFolder) { DeleteFolder(DelFolder, user_id); }
```

Comment:
I think, the permission check should be encapsulated in a function `check_permission_folder_del` similar to `check_permission_del` for uploads. (One might also rename the functions to be consistend)

Then one does not have to pass the `user_id`.